kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -341,6 +407,42 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    /// Check that a division does not overflow.
+    /// For integer types, division by zero is UB, as is MIN / -1 for signed.
+    /// Note that the compiler already inserts these checks for regular division.
```

Comment:
I don't see any test that demonstrates this. Can we include the checks coming from `rustc` into them so we can verify they still happen?

Also, can we refer to an issue tracking the work of getting rid of either?

Reply:
Yeah... Getting rid of them might be a bit tricky. But I can capture that in an issue if you prefer

Reply:
Duplicate checks don't hurt for now but will complicate our post-processing moving forward. So yes, I'd like to capture it - the solution isn't necessarily to get rid of them, but long term we should agree on a way to uniformly report them and consider sanity checks to cover corner cases.